# Oyster_OsHv
Mapping genes conferring resistance to OsHV-1 in the Pacific Oyster 
Crassostrea gigas 

Understanding the genetic basis of resistance to ostreid herpes virus 1 (OsHV-1) by the 
Pacific oyster is important for the global oyster industry. The virulent Var forms of OsHV-1 have 
caused devastating losses of farmed Pacific oysters in Western Europe, Australia, New Zealand, 
and Asia, and OsHV-1 µVar is expected eventually to arrive on the U.S. West Coast. Since 1993, 
OsHV-1 has caused mass mortalities of seed and juvenile Pacific oysters in Tomales Bay, CA, 
with losses ranging from 50% to 100% during the warmest summer months. This project, 
therefore, was designed to generate high-density linkage maps to detect QTL (quantitative trait 
loci) for resistance to OsHV-1 infection or mortality. Two full-sib families (1252 and 5819) 
from four unrelated parents were produced and planted in Tomales Bay before OsHV-1-caused 
mass mortalities in late September 2015. Over a two-week period, family 1252 experienced 55%, 
and family 5819, 16% mortality. Reduced representational genomic libraries were constructed 
for pre- and post-exposure samples (268 individuals from 1252 and 290 individuals from 5819) 
as well as the parents of the two families. Individual libraries were bar-coded and sequenced in 
pools, using genotyping-by-synthesis methods. Changes in genotypic frequencies particular to the 
exposure period suggest that seven viability QTLs in families 5819 and 1252, are associated 
with resistance to OsHV-1 and survival. Although the current state of the assembly of the Pacific 
oyster genome impedes identification of candidate genes, these findings strongly suggest that 
marker-assisted selection could be used to select oyster familyies resistant to OsHV-1 infection or 
mortality. 
